File: frontend/src/api/costs.ts
/**
 * Client API pour les endpoints de cost governance.
 */
import apiClient from './client'

export interface BudgetResponse {
  quota: number
  amount: number
  percentage: number
  remaining: number
}

export interface UpdateBudgetRequest {
  quota: number
}

export interface DailyCost {
  date: string
  cost: number
}

export interface UsageResponse {
  daily_costs: DailyCost[]
  total: number
  percentage: number
}

/**
 * Récupère le budget actuel.
 */
export async function getBudget(): Promise<BudgetResponse> {
  const response = await apiClient.get('/api/v1/costs/budget')
  return response.data
}

/**
 * Met à jour le quota budget.
 */
export async function updateBudget(quota: number): Promise<BudgetResponse> {
  const response = await apiClient.put('/api/v1/costs/budget', { quota })
  return response.data
}

/**
 * Récupère l'usage avec graphique (coûts quotidiens).
 */
export async function getUsage(): Promise<UsageResponse> {
  const response = await apiClient.get('/api/v1/costs/usage')
  return response.data
}

File: api/schemas/costs.py
"""Schémas Pydantic pour les endpoints de cost governance."""
from typing import List, Optional
from pydantic import BaseModel, Field, ConfigDict


class BudgetResponse(BaseModel):
    """Schéma de réponse pour le budget actuel."""
    quota: float = Field(..., ge=0.0, description="Quota mensuel en USD")
    amount: float = Field(..., ge=0.0, description="Montant dépensé en USD")
    percentage: float = Field(..., ge=0.0, le=100.0, description="Pourcentage du budget utilisé (0-100)")
    remaining: float = Field(..., ge=0.0, description="Montant restant en USD")
    
    model_config = ConfigDict()


class UpdateBudgetRequest(BaseModel):
    """Schéma de requête pour mettre à jour le quota budget."""
    quota: float = Field(..., ge=0.0, description="Nouveau quota mensuel en USD")
    
    model_config = ConfigDict()


class DailyCost(BaseModel):
    """Coût journalier."""
    date: str = Field(..., description="Date au format YYYY-MM-DD")
    cost: float = Field(..., ge=0.0, description="Coût total pour cette date en USD")
    
    model_config = ConfigDict()


class UsageResponse(BaseModel):
    """Schéma de réponse pour l'usage avec graphique."""
    daily_costs: List[DailyCost] = Field(..., description="Coûts quotidiens pour le graphique")
    total: float = Field(..., ge=0.0, description="Coût total du mois en USD")
    percentage: float = Field(..., ge=0.0, le=100.0, description="Pourcentage du budget utilisé")
    
    model_config = ConfigDict()
